server/src/main/java/org/elasticsearch/rest/action/search/SearchCapabilities.java

```diff
@@ -49,6 +49,8 @@ private SearchCapabilities() {}
     private static final String OPTIMIZED_SCALAR_QUANTIZATION_BBQ = "optimized_scalar_quantization_bbq";
     private static final String KNN_QUANTIZED_VECTOR_RESCORE = "knn_quantized_vector_rescore";
 
+    private static final String MAX_ANALYZED_OFFSET_DEFAULT = "max_analyzed_offset_default";
```

Comment:
would be nice to add highlight here to indicate it is relevant for highlighting:

    private static final String HIGHLIGHT_MAX_ANALYZED_OFFSET_DEFAULT = "highlight_max_analyzed_offset_default";